### DISCUSSION TITLE
run example found error

### DISCUSSION THREAD

**Discussion by wisonzhu** [argoproj/argo-workflows#7231] [Q&A] [answered]
# github.com/argoproj/argo-workflows/v3/pkg/apis/workflow/v1alpha1
../../../pkg/mod/github.com/argoproj/argo-workflows/v3@v3.2.3/pkg/apis/workflow/v1alpha1/openapi_generated.go:139:3: cannot use "github.com/go-openapi/spec".Schema{...} (type "github.com/go-openapi/spec".Schema) as type "k8s.io/kube-openapi/pkg/validation/spec".Schema in field value
../../../pkg/mod/github.com/argoproj/argo-workflows/v3@v3.2.3/pkg/apis/workflow/v1alpha1/openapi_generated.go:151:3: cannot use "github.com/go-openapi/spec".Schema{...} (type "github.com/go-openapi/spec".Schema) as type "k8s.io/kube-openapi/pkg/validation/spec".Schema in field value
../../../pkg/mod/github.com/argoproj/argo-workflows/v3@v3.2.3/pkg/apis/workflow/v1alpha1/openapi_generated.go:158:8: cannot use ref("github.com/argoproj/argo-workflows/v3/pkg/apis/workflow/v1alpha1.TarStrategy") (type "k8s.io/kube-openapi/pkg/validation/spec".Ref) as type "github.com/go-openapi/spec".Ref in field value
../../../pkg/mod/github.com/argoproj/argo-workflows/v3@v3.2.3/pkg/apis/workflow/v1alpha1/openapi_generated.go:163:8: cannot use ref("github.com/argoproj/argo-workflows/v3/pkg/apis/workflow/v1alpha1.NoneStrategy") (type "k8s.io/kube-openapi/pkg/validation/spec".Ref) as type "github.com/go-openapi/spec".Ref in field value
../../../pkg/mod/github.com/argoproj/argo-workflows/v3@v3.2.3/pkg/apis/workflow/v1alpha1/openapi_generated.go:168:8: cannot use ref("github.com/argoproj/argo-workflows/v3/pkg/apis/workflow/v1alpha1.ZipStrategy") (type "k8s.io/kube-openapi/pkg/validation/spec".Ref) as type "github.com/go-openapi/spec".Ref in field value
../../../pkg/mod/github.com/argoproj/argo-workflows/v3@v3.2.3/pkg/apis/workflow/v1alpha1/openapi_generated.go:181:3: cannot use "github.com/go-openapi/spec".Schema{...} (type "github.com/go-openapi/spec".Schema) as type "k8s.io/kube-openapi/pkg/validation/spec".Schema in field value
../../../pkg/mod/github.com/argoproj/argo-workflows/v3@v3.2.3/pkg/apis/workflow/v1alpha1/openapi_generated.go:200:11: cannot use ref("github.com/argoproj/argo-workflows/v3/pkg/apis/workflow/v1alpha1.Parameter") (type "k8s.io/kube-openapi/pkg/validation/spec".Ref) as type "github.com/go-openapi/spec".Ref in field value
../../../pkg/mod/github.com/argoproj/argo-workflows/v3@v3.2.3/pkg/apis/workflow/v1alpha1/openapi_generated.go:220:11: cannot use ref("github.com/argoproj/argo-workflows/v3/pkg/apis/workflow/v1alpha1.Artifact") (type "k8s.io/kube-openapi/pkg/validation/spec".Ref) as type "github.com/go-openapi/spec".Ref in field value
../../../pkg/mod/github.com/argoproj/argo-workflows/v3@v3.2.3/pkg/apis/workflow/v1alpha1/openapi_generated.go:280:8: cannot use ref("github.com/argoproj/argo-workflows/v3/pkg/apis/workflow/v1alpha1.S3Artifact") (type "k8s.io/kube-openapi/pkg/validation/spec".Ref) as type "github.com/go-openapi/spec".Ref in field value
../../../pkg/mod/github.com/argoproj/argo-workflows/v3@v3.2.3/pkg/apis/workflow/v1alpha1/openapi_generated.go:286:8: cannot use ref("github.com/argoproj/argo-workflows/v3/pkg/apis/workflow/v1alpha1.GitArtifact") (type "k8s.io/kube-openapi/pkg/validation/spec".Ref) as type "github.com/go-openapi/spec".Ref in field value
../../../pkg/mod/github.com/argoproj/argo-workflows/v3@v3.2.3/pkg/apis/workflow/v1alpha1/openapi_generated.go:286:8: too many errors

**Comment by terrytangyuan** [comment]
Which example? Probably you need to update your dependencies.

**Comment by alswl** [comment, marked as answer]
kube-openapi upgrade it's dependency, and the `spec` type is migrate from 'go-openapi/spec`  to `k8s.io/kube-openapi/pkg/validation/spec`.

```
	// not works for kyaml v0.11.0, but works for argo-cd
	//k8s.io/kube-openapi => k8s.io/kube-openapi v0.0.0-20210305001622-591a79e4bda7
	k8s.io/kube-openapi => k8s.io/kube-openapi v0.0.0-20210421082810-95288971da7e

```

The break change is:

```
kube-openapi (9528897) $ g remote -v
origin  git@github.com:kubernetes/kube-openapi.git (fetch)
origin  git@github.com:kubernetes/kube-openapi.git (push)
kube-openapi (9528897) $ g co 591a79e4bda7
Previous HEAD position was 9528897 Merge pull request #234 from gautierdelorme/add_non_schema_types_to_validation_spec
HEAD is now at 591a79e Merge pull request #230 from liggitt/deepequal-value
kube-openapi (591a79e) $ c pkg/common/common.go G '/spec'
        "github.com/go-openapi/spec"
kube-openapi (591a79e) $ g co 95288971da7e
Previous HEAD position was 591a79e Merge pull request #230 from liggitt/deepequal-value
HEAD is now at 9528897 Merge pull request #234 from gautierdelorme/add_non_schema_types_to_validation_spec
kube-openapi (9528897) $ c pkg/common/common.go G '/spec'
        "k8s.io/kube-openapi/pkg/validation/spec"
```

It's a break changes. It seems only way is update argo project's `go.mod`.
